src/controllers/user.js

```diff
@@ -26,10 +26,16 @@ const getPreferences = (req, res) => {
 const updatePreferences = (req, res) => {
   try {
     const { preferences } = req.body;
-    const { id: userId } = req.user || {};
+    const { id: userId, preferences: existingPreferences } = req.user || {};
+
+    if (!preferences || !Array.isArray(preferences) || !preferences?.length)
```

Comment:
You can think about putting this in a variable for readability instead of having 3 conditions